src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/TypeSymbolExtensions.cs

```diff
@@ -300,5 +300,25 @@ public static (ImmutableArray<ITypeSymbol> TypeArguments, ImmutableArray<Nullabl
             }
             return (typeArguments.ToImmutable(), nullableAnnotations.ToImmutable());
         }
+
+        /// <summary>
+        /// Returns if the type is of the string representation of a type
+        /// </summary>
+        public static bool IsOfType(this INamedTypeSymbol type, string typeName)
```

Comment:
Should we move all of our type-name checks to use this helper instead of comparing with `ToDisplayString()`?

Reply:
We can leave this for a future PR.